fineract-provider/src/main/java/org/apache/fineract/portfolio/client/service/ClientReadPlatformService.java

```diff
@@ -60,4 +60,7 @@ public interface ClientReadPlatformService {
     LocalDate retrieveClientTransferProposalDate(Long clientId);
 
     void validateClient(Long clientId);
+
+    Long retrieveLoanIdByExternalId(String externalId);
```

Comment:
This is misplaced i believe. The Client service should not resolve Loan related data...

Reply:
Done updated to `retrieveClientIdByExternalId`